diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index 3445d6b3e..65fa0e30e 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -69,6 +69,7 @@
 #include "World.h"
 #include "WorldPacket.h"
 #include <cmath>
+#include "../../../modules/mod-spell-regulator/src/SpellRegulator.h
 
 float baseMoveSpeed[MAX_MOVE_TYPE] =
 {
@@ -825,6 +826,11 @@ uint32 Unit::DealDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage
     // Hook for OnDamage Event
     sScriptMgr->OnDamage(attacker, victim, damage);
 
+    if ((damagetype == SPELL_DIRECT_DAMAGE || damagetype == DOT) && spellProto)
+    {
+        sSpellRegulator->Regulate(damage, spellProto->Id);
+    }
+
     if (victim->IsPlayer() && attacker != victim)
     {
         // Signal to pets that their owner was attacked
@@ -6363,6 +6369,7 @@ void Unit::SendSpellNonMeleeReflectLog(SpellNonMeleeDamage* log, Unit* attacker)
 
 void Unit::SendSpellNonMeleeDamageLog(SpellNonMeleeDamage* log)
 {
+    sSpellRegulator->Regulate(log->damage, log->spellInfo->Id);
     WorldPacket data(SMSG_SPELLNONMELEEDAMAGELOG, (16 + 4 + 4 + 4 + 1 + 4 + 4 + 1 + 1 + 4 + 4 + 1)); // we guess size
     //IF we are in cheat mode we swap absorb with damage and set damage to 0, this way we can still debug damage but our hp bar will not drop
     uint32 damage = log->damage;
@@ -6443,6 +6450,7 @@ void Unit::ProcDamageAndSpell(Unit* actor, Unit* victim, uint32 procAttacker, ui
 void Unit::SendPeriodicAuraLog(SpellPeriodicAuraLogInfo* pInfo)
 {
     AuraEffect const* aura = pInfo->auraEff;
+    sSpellRegulator->Regulate(pInfo->damage, aura->GetId());
     WorldPacket data(SMSG_PERIODICAURALOG, 30);
     data << GetPackGUID();
     data << aura->GetCasterGUID().WriteAsPacked();
diff --git a/src/server/scripts/Commands/cs_reload.cpp b/src/server/scripts/Commands/cs_reload.cpp
index 45434d502..2186b3104 100644
--- a/src/server/scripts/Commands/cs_reload.cpp
+++ b/src/server/scripts/Commands/cs_reload.cpp
@@ -48,6 +48,7 @@ EndScriptData */
 #include "WardenCheckMgr.h"
 #include "WaypointMgr.h"
 #include "WorldGlobals.h"
+#include "../../../modules/mod-spell-regulator/src/SpellRegulator.h"
 
 using namespace Acore::ChatCommands;
 
@@ -164,6 +165,7 @@ public:
             { "spell_pet_auras",               HandleReloadSpellPetAurasCommand,              SEC_ADMINISTRATOR, Console::Yes },
             { "spell_proc_event",              HandleReloadSpellProcEventCommand,             SEC_ADMINISTRATOR, Console::Yes },
             { "spell_proc",                    HandleReloadSpellProcsCommand,                 SEC_ADMINISTRATOR, Console::Yes },
+            { "spell_regulator",			   HandleReloadSpellRegulator,                    SEC_ADMINISTRATOR, Console::Yes },
             { "spell_scripts",                 HandleReloadSpellScriptsCommand,               SEC_ADMINISTRATOR, Console::Yes },
             { "spell_target_position",         HandleReloadSpellTargetPositionCommand,        SEC_ADMINISTRATOR, Console::Yes },
             { "spell_threats",                 HandleReloadSpellThreatsCommand,               SEC_ADMINISTRATOR, Console::Yes },
@@ -222,6 +224,7 @@ public:
         HandleReloadMotdCommand(handler);
         HandleReloadBroadcastTextCommand(handler);
         HandleReloadBattlegroundTemplate(handler);
+        HandleReloadSpellRegulator(handler);
         return true;
     }
 
@@ -947,6 +950,15 @@ public:
         return true;
     }
 
+    static bool HandleReloadSpellRegulator(ChatHandler * handler)
+    {
+        #define sSpellRegulator SpellRegulator::instance()
+
+        sSpellRegulator->LoadFromDB();
+        handler->SendGlobalGMSysMessage("DB table `spell_regulator` reloaded.");
+        return true;
+    }
+
     static bool HandleReloadSpellPetAurasCommand(ChatHandler* handler)
     {
         LOG_INFO("server.loading", "Reloading Spell pet auras...");
